### PR TITLE
Optimise forms

### PR DESCRIPTION
**Pull request by ImemBA1**

### CODE REVIEW

File: gestionnaire_de_stage_frontend/src/services/user-service.js [hunk 1]
```diff
@@ -6,6 +6,18 @@ export async function getSupervisors() {
     return await response.json();
 }
 
+export async function changePassword(typeUser, id, password) {
+    return await fetch(`${urlBackend}/${typeUser}/change_password/${id}`, requestInit(methods.POST, password)).then(response => {
```
Comment (OussaKab) [inline gestionnaire_de_stage_frontend/src/services/user-service.js:10]:
```suggestion
    return await fetch(`${urlBackend}/${typeUser}/change_password/${id}`, requestInit(methods.PUT, password)).then(response => {
```

**Review by OussaKab — CHANGES_REQUESTED** [review]

File: backend/src/main/java/com/gestionnaire_de_stage/service/OfferApplicationService.java [hunk 2]
```diff
@@ -96,14 +97,25 @@ private boolean isDateInvalid(LocalDateTime date) {
...
+    public List<Student> getOffersApplicationsStageTrouver() {
+        List<OfferApplication> offerApplicationList = offerApplicationRepository.getAllByStatusAndSession_YearGreaterThanEqualAndCurriculum_Student_SupervisorIsNull(Status.STAGE_TROUVE, Year.now());
+
+        return offerApplicationList.stream()
+                .map(offerApplication -> getStudent(offerApplication.getCurriculum())).distinct().collect(Collectors.toList());
+    }
+
+    private Student getStudent(Curriculum curriculum) {
+        return curriculum != null ? curriculum.getStudent() : null;
+    }
```
[the remark this answers is not in the document]
Comment (ImemBA1) [reply backend/src/main/java/com/gestionnaire_de_stage/service/OfferApplicationService.java:117]: non

File: gestionnaire_de_stage_frontend/src/components/Offers/OfferView.jsx [hunk 3]
```diff
@@ -1,24 +1,46 @@
...
+                    <div className={styles.cardJobDetails}>
+                        <p className={`d-flex align-items-center ${styles.cardCompanyLocation}`}>
+                            <MdLocationPin/> {offer.address}
+                        </p>
+                        <p className={`d-flex align-items-center ${styles.cardJobDuration}`}>
+                            <BsClock title="Durée du stage" className={"me-1"}/> {offer.nbSemaine}
+                        </p>
+                        <p className={`d-flex align-items-center ${styles.cardListingDate}`}>
+                            <BsClockHistory className={"me-1"}/> Il y
+                            a {Math.ceil((new Date().getTime() - new Date(offer.created).getTime()) / (1000 * 3600 * 24))} jour(s)
```
[the remark this answers is not in the document]
Comment (ImemBA1) [reply gestionnaire_de_stage_frontend/src/components/Offers/OfferView.jsx:27]: qwpe[jhugqhhehgeibpoijhgrpoikljp;oirhg[rnwsth
]

**Review by FrostDistrict — APPROVED** [review]
so many lines jeeeeezz

**Review by BertSa — APPROVED** [review]

**Review by BertSa — APPROVED** [review]

**Pull request merged**